### PR TITLE
Merge develop into main

### PR DESCRIPTION
**Pull request by henneberger**

### CODE REVIEW

**Review by mbroecheler — CHANGES_REQUESTED** [review]
Looking good. The simplification around SQRLTable is very noticable. Good work!

File: sqrl-planner/sqrl-common/src/main/java/com/datasqrl/calcite/QueryPlanner.java [hunk 1]
```diff
@@ -71,7 +71,8 @@
 @Getter
 public class QueryPlanner {
 
-  private static final AtomicInteger uniqueCompilerId = new AtomicInteger(0);
+  private final AtomicInteger uniqueCompilerId = new AtomicInteger(0);
+  private final AtomicInteger uniquePkId = new AtomicInteger(0);
```
[the remark this answers is not in the document]
Comment (henneberger) [reply sqrl-planner/sqrl-common/src/main/java/com/datasqrl/calcite/QueryPlanner.java:75]: It alleviates the need to keep track of the table alias when adding columns to the select/group/order clause.

File: sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/flink/FlinkDebugPhysicalIT/debugC3602OutputSelectTablesTest.txt [hunk 2]
```diff
@@ -1,186 +1,183 @@
...
+2023-05-19 02:16:39.553, 1000121, lalelu@hottunes.org, Michelle Dagnes, 166.5, 1
+2023-05-19 02:28:39.553, 1000107, emily.ludies@hotmail.com, Emily F. Ludies, 146.82, 2
 >>>order_again-data
-1332, 8, 1, 2023-05-19T02:28:39.553-07:00
-3225, 1, 1, 2023-05-19T02:16:39.553-07:00
-3571, 2, 2, 2023-05-19T02:28:39.553-07:00
-7235, 1, 1, 2023-05-19T01:29:39.553-07:00
-7552, 3, 1, 2023-05-19T02:16:39.553-07:00
-8757, 2, 1, 2023-05-19T01:29:39.553-07:00
+1332, 8, 1, 2023-05-19 02:28:39.553
```
[the remark this answers is not in the document]
Comment (henneberger) [reply sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/flink/FlinkDebugPhysicalIT/debugC3602OutputSelectTablesTest.txt:175]: we moved off h2 completely for tests

File: sqrl-engines/sqrl-engine-core/src/main/java/com/datasqrl/plan/hints/TimestampHint.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import lombok.Getter;
+import org.apache.calcite.rel.hint.RelHint;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Getter
+@AllArgsConstructor
+public class TimestampHint implements SqrlHint {
```
[the remark this answers is not in the document]
Comment (henneberger) [reply sqrl-engines/sqrl-engine-core/src/main/java/com/datasqrl/plan/hints/TimestampHint.java:17]: Only adding the hint was scoped in the sprint

File: sqrl-planner/sqrl-common/src/main/java/com/datasqrl/calcite/QueryPlanner.java [hunk 4]
```diff
@@ -410,7 +411,11 @@ public RelNode run(RelNode relNode, RelRule... rules) {
   }
 
   public String relToString(Dialect dialect, RelNode relNode) {
-    return sqlToString(dialect, relToSql(dialect, relNode));
+    String sql = sqlToString(dialect, relToSql(dialect, relNode));
+    if (dialect == Dialect.POSTGRES) {
```
[the remark this answers is not in the document]
Comment (henneberger) [reply sqrl-planner/sqrl-common/src/main/java/com/datasqrl/calcite/QueryPlanner.java:415]: good catch

File: sqrl-planner/sqrl-common/src/main/java/com/datasqrl/calcite/SqrlFramework.java [hunk 5]
```diff
@@ -25,7 +29,9 @@ public class SqrlFramework {
   private RelMetadataProvider relMetadataProvider;
   private AtomicInteger uniqueMacroInt = new AtomicInteger(0);
   private AtomicInteger uniqueTableInt = new AtomicInteger(0);
+  private Map<Name, AtomicInteger> tableNameToIdMap = new HashMap<>();
   private AtomicInteger uniqueColumnInt = new AtomicInteger(0);
+  private Map<Name,AtomicInteger> tableNameShadowing = new HashMap<Name,AtomicInteger>();
```
[the remark this answers is not in the document]
Comment (henneberger) [reply sqrl-planner/sqrl-common/src/main/java/com/datasqrl/calcite/SqrlFramework.java:34]: not used, I'll clean up all unique counters

File: sqrl-planner/sqrl-common/src/main/java/com/datasqrl/calcite/schema/ScriptPlanner.java [hunk 6]
```diff
@@ -630,6 +621,12 @@ public Result visitSetOperation(SqlCall node, Context context) {
     }
   }
 
+  @Value
+  public static class PullupColumn {
+    String columnName;
```
[the remark this answers is not in the document]
Comment (henneberger) [reply sqrl-planner/sqrl-common/src/main/java/com/datasqrl/calcite/schema/ScriptPlanner.java:626]: One is final display name, the other is a globally unique name b/c there is no guarantee there is a table alias on the table macro

File: sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/flink/FlinkDebugPhysicalIT/debugC3602OutputSelectTablesTest.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (mbroecheler) [reply sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/flink/FlinkDebugPhysicalIT/debugC3602OutputSelectTablesTest.txt:175]: Ah, I didn't realize that. Does Postgres not support timezones? Is this an issue we need to address?

[the remark this answers is not in the document]
Comment (henneberger) [reply sqrl-testing/sqrl-integration-tests/src/test/resources/snapshots/com/datasqrl/flink/FlinkDebugPhysicalIT/debugC3602OutputSelectTablesTest.txt:175]: It is a `timestamp with time zone` type. From the postgres docs:

For timestamp with time zone, the internally stored value is always in UTC (Universal Coordinated Time, traditionally known as Greenwich Mean Time, GMT). An input value that has an explicit time zone specified is converted to UTC using the appropriate offset for that time zone. If no time zone is stated in the input string, then it is assumed to be in the time zone indicated by the system's [TimeZone](https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-TIMEZONE) parameter, and is converted to UTC using the offset for the timezone zone.

**Review by mbroecheler — APPROVED** [review]

**Pull request merged**